app.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+import logging
+import os
+import typing
+
+import boto3
+from django.db import transaction  # type: ignore
+from openstates.cli.reports import generate_session_report
+
+
+logger = logging.getLogger("openstates")
+s3_client = boto3.client('s3')
+
+
+def process_upload_function(event, context):
+    """
+    Process a file upload.
+    """
+
+    # Get the uploaded file's information
+    bucket = event['Records'][0]['s3']['bucket']['name']  # Will be `my-bucket`
+    key = event['Records'][0]['s3']['object']['key']  # Will be the file path of whatever file was uploaded.
+
+    # Get the bytes from S3
+    datadir = "/tmp/"
+
+    state_acronym = key.split("/")[0]  # e.g az, al, etc
```

Comment:
Thanks @johnseekins - this makes sense, I've updated this.

Possibility of extracting from metadata? Not at the moment because this metadata is default meta data that AWS picks up from  the event of file upload, I can confirm if we can do anything flexible around the metadata